### ISSUE TITLE
FeatureImp not working for logLoss and mse loss with a classification model

### ISSUE THREAD

**Issue by jasondmorgenstern** [christophM/iml#101]
Hi, 

Thank you for a great package. I'm just running into an issue when trying to use the FeatureImp function with either a "logLoss" or "mse" loss. I don't want to use classification error or F1.

I'm working on a closed, secure system so I'm using R version 3.5.3. "iml" is version 0.9.0. I can't share the actual dataset I'm using so I reproduced it using a caret dataset:

`library(caret)
library(mlr)
library(iml)

data(GermanCredit)

CIF <- makeLearner(cl = "classif.cforest", predict.type = "prob")
Task <- makeClassifTask(data = GermanCredit, target = "Class", positive = "Good")
mod <- train(CIF, Task)

pred <- Predictor$new(model = mod, data = GermanCredit, y = "Class", type = "prob")

FI <- FeatureImp$new(pred, loss = "logLoss")`

**Error:**

`Error in Math.factor(predicted) : ‘log’ not meaningful for factors`

And with an "mse" loss:

`FI <- FeatureImp$new(pred, loss = "mse")`

**Error:**

`Error in if (self$original.error == 0) { : 
  missing value where TRUE/FALSE needed
In addition: Warning message:
In Ops.factor(actual, predicted) : ‘-’ not meaningful for factors`

Thanks in advance for any help.

**Comment by christophM** [comment]
The problem is that GermanCredit$Class is a factor.

The example works when you do `pred <- Predictor$new(model = mod, data = GermanCredit, y = GermanCredit$Class == "Good", type = "prob", class = "Good")`

But that is not very obvious. I think the best would be if these things are automatically determined based on the chosen loss function.
For example, if "logLoss" is chosen, the correct y column should be chosen.

**Comment by jasondmorgenstern** [comment]
Thank you Christoph! The consideration and help is much appreciated.
